### ISSUE TITLE
Unknown Finish Reason

### ISSUE THREAD

**Issue by alexandergagliano** [rgbkrk/chatlab#127]
-   ChatLab version: 1.3.0
-   Python version: 3.10.12
-   Operating System: CentOS

### Description

Hi all! I'm trying to run the tutorial from functionary [here](https://functionary.meetkai.com/automatic-tool-execution) after launching a vLLM server.  The code returns: 

```
INFO:     127.0.0.1:48176 - "POST /v1/chat/completions HTTP/1.1" 200 OK
UNKNOWN FINISH REASON: 'tool_calls'. If you see this message, report it as an issue to https://github.com/rgbkrk/chatlab/issues
USER: What is the price of the car named 'Rhino'?
```

Any thoughts on what could be the problem? Thanks! 

### What I Did

```
import chatlab
import asyncio

def get_car_price(car_name: str):
    """this function is used to get the price of the car given the name
    :param car_name: name of the car to get the price
    """
    car_price = {
        "rhino": {"price": "$20000"},
        "elephant": {"price": "$25000"}
    }
    for key in car_price:
        if key in car_name.lower():
            return {"price": car_price[key]}
    return {"price": "unknown"}

chat = chatlab.Chat(model="meetkai/functionary-small-v2.2", base_url="http://localhost:8000/v1", api_key="functionary")
chat.register(get_car_price)
asyncio.run(chat.submit("What is the price of the car named 'Rhino'?", stream=False))

for message in chat.messages:
    role = message["role"].upper()
    if "function_call" in message:
        func_name = message["function_call"]["name"]
        func_param = message["function_call"]["arguments"]
        print(f"{role}: call function: {func_name}, arguments:{func_param}")
    else:
        content = message["content"]
        print(f"{role}: {content}")

```

**Comment by rgbkrk** [comment]
Thanks for reporting. I'm working on this in https://github.com/rgbkrk/chatlab/pull/125 and will try to get it finished up over the next few weeks.

**Comment by rgbkrk** [comment]
Done in #131. [Shipped in v2.0.0](https://github.com/rgbkrk/chatlab/releases/tag/v2.0.0).
